src/main.cxx

```diff
@@ -435,6 +435,7 @@ for(time_step_num=0;time_step_num<num_time_steps;time_step_num++)
       printf(" ************ time step %d ******** \n",time_step_num);
     }
     
+    if( state_previous != NULL ) listDelete(state_previous);
     state_previous = NULL;
     state_previous = listCopy(head);
```

Comment:
The condition is redundant with the implementation of `listDelete`, and the repeat assignment is pointless.
```suggestion
    listDelete(state_previous);
    state_previous = listCopy(head);
```

Reply:
I left that to be as "close as possible to the original code", but it really isn't a significant difference so may as well simplify.